wire-ios-transport/WireTransport.docc/Certificate Pinning.md

```diff
@@ -0,0 +1,23 @@
+# Certificate pinning
+
+Verify identity of Backend servers by checking certificates.
+
+## Overview
+
+In order to make sure we're talking to the right server, we use certificate pinning. This means that we check the certificate of the server we're talking to against a known certificate. If the certificates match, we know we're talking to the right server.
+
+
+Here, we only verify the public key of the certificate that way when server changes its certificate (and keeping the same public key), we won't have to update all clients. We also check the certificate downloaded from the server is not expired.
```

Comment:
```suggestion
One approach of certificate pinning involves comparing the entire certificate. The downside to this is that when the server eventually updates their certificate, then also the client must update its local certificate to continue to trust that server. This, in effect, means that older clients will no longer no longer trust the server.

To avoid this issue, Wire employs a second approach, namely to verify only the public key of the certificate. In this way, if the server renews its certificate with the same public-private key pair, then all existing clients that have the same public key will continue to trust the server, even with new certificates. Specifically, a server is considered trusted when its certificate is not expired and its public key matches the known and trusted public key.
```